src/senaite/patient/browser/patientfolder.py

```diff
@@ -131,6 +141,12 @@ def to_utf8(self, s):
         """
         return api.safe_unicode(s).encode("utf8")
 
+    def translate(self, message):
+        """Returns the translated message
+        """
+        ts = api.get_tool("translation_service")
+        return ts.translate(message)
```

Comment:
I think you should pass in here the `context` and `request` as well to make it work in different languages.
Also see `Products.CMFPlone.TranslationServiceTool.TranslationServiceTool.translate`

Reply:
Note that Plone's ts ultimately relies on Zope's i18n `translate` function. And if the message passed in is from type `Message`, i18n gives priority to the domain and other info set at `Message` level, regardless of what you pass through kwargs: https://github.com/zopefoundation/zope.i18n/blob/bb0b8d2ee88a1e280cc86d858e7a2524ba0dc8d9/src/zope/i18n/__init__.py#L164-L170

Since we always rely on our own add-on `MessageFactory`, the domain is always assigned to the message and therefore, there is no need to explicitely pass the domain. However, I agree is always a good idea to be consistent, so I added a `translate` function (that we could eventually port to `senaite.core` and replace the existing `utils.t`): https://github.com/senaite/senaite.patient/pull/84/commits/3c6d98a03ee4b991db9d85a80d9eeea02d01cae1

Reply:
> [..] that we could eventually port to senaite.core and replace the existing utils.t): https://github.com/senaite/senaite.patient/commit/3c6d98a03ee4b991db9d85a80d9eeea02d01cae1

Done with https://github.com/senaite/senaite.core/pull/2389